src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```
@@ -286,6 +287,30 @@ public async Task HandleGenerateTSqlModelRequest(GenerateTSqlModelParams request
             }
         }
 
+        /// <summary>
+        /// Handles request to get objects from sql model
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleGetObjectsFromTSqlModelRequest(GetObjectsFromTSqlModelParams requestParams, RequestContext<TSqlObjectInfo[]> requestContext)
+        {
+            try
+            {
+                TSqlObjectInfo[] objectInfos = {};
+                var model = projectModels.Value[requestParams.ProjectUri];
+
+                if (model != null)
```

We should throw an error if there isn't a model